test/integration/scheduler_perf/scheduler_perf_test.go

```diff
@@ -363,15 +363,7 @@ func runWorkload(b *testing.B, tc *testCase, w *workload) []DataItem {
 			if err := createPods(namespace, concreteOp, clientset); err != nil {
 				b.Fatalf("op %d: %v", opIndex, err)
 			}
-			if concreteOp.SkipWaitToCompletion {
-				// Only record those namespaces that may potentially require barriers
-				// in the future.
-				if _, ok := numPodsScheduledPerNamespace[namespace]; ok {
-					numPodsScheduledPerNamespace[namespace] += concreteOp.Count
-				} else {
-					numPodsScheduledPerNamespace[namespace] = concreteOp.Count
-				}
```

Comment:
since we don't have any "barrier" Op defined in the template, those code could be simply removed without any issue, but if we defined a "barrier" Op in the template, suppose there is any real usecase for this, the test will be crashed always, see code at line 385
```
				if _, ok := numPodsScheduledPerNamespace[namespace]; !ok {
					b.Fatalf("op %d: unknown namespace %s", opIndex, namespace)
				}
```

Reply:
I am not quite understanding what's the actual meaning of this `Op`, this `Op` must be explicitly set in the template, give a `skipWaitToCompletion` defined to `true` without the `Op` doesn't bring us anything, e.g. the `Unschedulable` benchmark, what am I missing? @adtac 
```
// A barrierOp is added at the end of each each workload automatically.
```

Reply:
>the test will be crashed always, see code at line 385

yes, we shouldn't remove the `numPodsScheduledPerNamespace` segment. Instead, the real problem seems to be this, as @chendave pointed out:

>// A barrierOp is added at the end of each each workload automatically.

The `waitUntilPodsScheduled` call at the end of the `runWorkload` function seems to the root cause of this bug and it should be removed. I had incorrectly assumed that all must be scheduled at the end of the workload.

Reply:
Thanks @chendave and @adtac. +1 to remove the assumption that each workload needs to be followed by a `waitUntilPodsScheduled`.

So closing this issue in favor of #98721.